### DISCUSSION TITLE
Installation freezes at "Creating an optimized production build..."

### DISCUSSION THREAD

**Discussion by B1ack1c3** [Chia-Network/chia-blockchain#1394] [Q&A]
Hi,

I am trying to install CHIA on Ubuntu 20.04 LTS (RAM 1GB + 7GB SWAP + HDD 400Go), the system seems to be frozen at this step "install-gui.sh" with this message :
"Creating an optimized production build..."

Thanks for your help

**Comment by dorkmo** [comment]
this can take a while, how long has it run?

**Reply by B1ack1c3** [reply]
Thanks a lot, I waited 4 hours and it's good now.
But I have a new error: 
/chia-blockchain/chia-blockchain-gui/node_modules/electron/dist/electron: error while loading shared libraries: libnss3.so: cannot open shared ob                                                                                                             ject file: No such file or directory
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! chia-blockchain@ electron: `electron .`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the chia-blockchain@ electron script.
npm ERR! This is probably not a problem with npm. There is likely additional log                                                                                                             ging output above.

**Reply by B1ack1c3** [reply]
I solved this error with the following command:
sudo apt-get install libnss3-dev
I have some others I'll keep you posted...

**Reply by B1ack1c3** [reply]
Some intersting tips here for missing dependeces  :

https://gist.github.com/winuxue/cfef08e2f5fe9dfc16a1d67a4ad38a01

And also this :

sudo apt-get install -y libgdk-pixbuf2.0-dev
sudo apt-get install -y libgbm-dev

**Reply by B1ack1c3** [reply]
To sum up, here it is the missing commands :
sudo apt-get install libnss3
sudo apt-get install libatk-bridge2.0-0
sudo apt-get install -y libgdk-pixbuf2.0-dev
sudo apt install libgtk-3-0
sudo apt-get install -y libgbm-dev

After that I see this message : 

/home/user/chia-blockchain/app.asar.unpacked/daemon
false
Error loading config - using defaults
/home/user/chia-blockchain/chia-blockchain-gui/node_modules/electron/dist/electron exited with signal SIGSEGV
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! chia-blockchain@ electron: `electron .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the chia-blockchain@ electron script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Reply by hoffmang9** [reply]
A couple things here. Some people try to run this as root and Electron will not allow that. A surprisingly large number of people try to start the GUI on machines whose OSes are headless and don't have Desktop or equivalent GUI stacks. Examples include trying to run in Azure or AWS where the runners do not support desktop (easily)

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
